### ISSUE TITLE
ImageProcessor.Web - Can't make it intercept a IHttpHandler request

### ISSUE THREAD

**Issue by sprintstar** [JimBobSquarePants/ImageProcessor#382]
I am trying to make ImageProcessor intercept an image url that's not just a static image.  I have an IHttpHandler that returns an image, but putting ImageProcessor.Web parameters on that url doesn't work, the image doesn't get processed.  For example:

// Image gets processed, and return 100px wide
http://localhost:57944/images/img.jpg?width=100

// Image doesn't get processed
http://localhost:57944/Logic/Image.ashx?src=/images/img.jpg&width=100

Is it possible to configure ImageProcessor to intercept those urls, or can it only intercept static file requests?

**Comment by JimBobSquarePants** [comment]
Hey there,

Use a custom IImageService instead as per the documentation. If you check out the code in the Framework branch you'll see examples.

Hope that helps.

**Comment by sprintstar** [comment]
Hello, thanks. I have tried writing my own, but am still having some problems.  If I wish to use something like
/MyImageHandler.axd?id=nnn&width=200

(where nnn is the id of an image)

.. and have image processor intercept that call, get the image and process it to, I end up writing code that goes to /MyImageHandler.axd to get the image.  This causes the image processor to intercept it, and we get a loop.

Also IsValidRequest gets passed "/MyImageHandler.axd" and so does GetImage, so i'm unclear how to make this work.  Please advise..

Basically we're using our own umbraco IFileSystem to serve images from a database, rather than the file system.  The /MyImageHandler.axd would be the basis of all our media image urls.

Many thanks

**Comment by JimBobSquarePants** [comment]
Ah right, I must ask, why the DB and not something like AzureBlobStorage? Storing images in a DB is never wise.

I have a very battle-tested working implementation (It's used in UAAS)

https://github.com/JimBobSquarePants/UmbracoFileSystemProviders.Azure

Feel free to borrow code from there if you are committed to using the DB.

So is your `IFileSystem` implementation still tied to a `HttpHandler`? That's not going to work well if it is.

In your `IImageService.GetImage(object Id)` implementation you will need to call the `IFileSystem.OpenFile(string path)` method to return the `byte[]`.

You should be able to retrieve a current instance of you `IFileSystem` implementation by using the following.

``` C#
IFileSystem fileSystem = FileSystemProviderManager.Current
                                                  .GetUnderlyingFileSystemProvider("YourMediaRoute");
```

Does that help?

**Comment by sprintstar** [comment]
Thank you for taking the time to respond.  We're writing an IFileSystem that will store the media in the database, but will utilize FILESTREAM or FileTable technology for speed. Storing media on the filesystem is a real pain point for our development team.  Here is our MK1 database IFileSystem: https://github.com/sprintstar/UmbracoFileSystemProviders.SQLFileStream

Anyway, so your recommendation is to intercept the handler, but not use it all all for getting the actual image, but instead go straight to the IFileSystem to get the image? Your sample code looks very helpful!

**Comment by JimBobSquarePants** [comment]
No worries :smile: 

My recommendation is not to use a `HttpHandler` at all.

You should be able to use your `IFileSystem` implementation with a virtual path provider as I do in the Azure version. That way your paths should follow the expected Umbraco default of `/media/1234/your_image.jpg`, will work with the cropper, and will be easily transferable amongst deployment environments. 

Grab my code from the repo and use anything you want from it in yours. None of it is crazy complicated so it should be easy enough to implement.
